File: 프로그래머스/0/181838. 날짜 비교하기/날짜 비교하기.js
function solution(date1, date2) {
    // year, month, day를 차례로 비교합니다.
    if (date1[0] !== date2[0]) {
        return date1[0] < date2[0] ? 1 : 0;
    }
    if (date1[1] !== date2[1]) {
        return date1[1] < date2[1] ? 1 : 0;
    }
    if (date1[2] !== date2[2]) {
        return date1[2] < date2[2] ? 1 : 0;
    }
    return 0; 
}

File: 프로그래머스/0/181838. 날짜 비교하기/날짜 비교하기.py
def solution(date1, date2):
    if date1[0] != date2[0]:
        return 1 if date1[0] < date2[0] else 0
    elif date1[1] != date2[1]:
        return 1 if date1[1] < date2[1] else 0
    elif date1[2] != date2[2]:
        return 1 if date1[2] < date2[2] else 0
    else:
        return 0